driver/src/main/java/org/neo4j/driver/internal/logging/JULogger.java

```diff
@@ -26,21 +26,22 @@
 public class JULogger implements Logger
 {
     private final java.util.logging.Logger delegate;
-    private final boolean debugEnabled;
 
     public JULogger( String name )
     {
         delegate = java.util.logging.Logger.getLogger( name );
-        debugEnabled = delegate.isLoggable( Level.FINE );
```

Comment:
You could investigate calling through MethodHandles from MutableCallsites. I think they're able to inline (if the MethodHandle reference is static final) so the cost of changing the log level (which is a quite rare event) would be a safe-point for deoptimising. I'm not completely sure if this is true, though.